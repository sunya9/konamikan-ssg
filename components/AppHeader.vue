<template>
  <header key="header">
    <div
      class="hero is-medium fix-background"
      :style="{
        'background-image': `url(${cover || defaultCover})`
      }"
    >
      <div class="hero-head">
        <nav-bar :home-button="homeButton" />
      </div>
      <div class="hero-body fix-z-index">
        <slot>
          <div class="container has-text-centered">
            <h1 class="title is-1 has-text-white">
              {{ title }}
            </h1>
            <h2 class="subtitle has-text-white-bis">
              {{ description }}
            </h2>
          </div>
        </slot>
      </div>
      <div v-if="$slots.footer" class="hero-foot">
        <slot name="footer" />
      </div>
    </div>
  </header>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'nuxt-property-decorator'
import NavBar from '~/components/NavBar.vue'

@Component({
  components: {
    NavBar
  }
})
export default class Header extends Vue {
  @Prop({ type: String, required: false, default: '' })
  cover!: string

  @Prop({ type: String, required: false, default: '' })
  title!: string

  @Prop({ type: String, required: false, default: '' })
  description!: string

  @Prop({ type: Boolean, required: false, default: false })
  homeButton!: boolean

  defaultCover = this.$resolveUrl(this.$setting.cover_image)

  get isIndex(): boolean {
    return this.$route.name === 'index'
  }
}
</script>
<style scoped>
.fix-background {
  background-position: center;
  background-size: cover;
  position: relative;
}
.fix-background::after {
  content: '';
  display: block;
  background-color: rgba(0, 0, 0, 0.3);
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  backdrop-filter: blur(4px);
}
.fix-z-index {
  z-index: 2;
}
</style>
