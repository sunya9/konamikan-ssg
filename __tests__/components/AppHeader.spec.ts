import { mount } from '@vue/test-utils'
import AppHeader from '@/components/AppHeader.vue'

describe('AppHeader', () => {
  it('normal', () => {
    expect(
      mount(AppHeader, {
        propsData: {
          title: 'title',
          description: 'description'
        }
      }).element
    ).toMatchSnapshot()
  })

  it('with home button', () => {
    expect(
      mount(AppHeader, {
        propsData: {
          homeButton: true
        }
      }).element
    ).toMatchSnapshot()
  })

  it('override default slot', () => {
    expect(
      mount(AppHeader, {
        slots: {
          default: 'default slot'
        }
      }).element
    ).toMatchSnapshot()
  })

  it('put footer slot', () => {
    expect(
      mount(AppHeader, {
        slots: {
          footer: 'footer'
        }
      }).element
    ).toMatchSnapshot()
  })

  it('override cover', () => {
    expect(
      mount(AppHeader, {
        propsData: {
          cover: 'passCoverViaProps.png'
        }
      }).element
    ).toMatchSnapshot()
  })
})
