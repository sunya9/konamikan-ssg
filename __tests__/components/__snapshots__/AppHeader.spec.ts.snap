// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader normal 1`] = `
<header
  class="hero is-medium fix-background"
  style="background-image: url(cover.png);"
>
  <div
    class="hero-head"
  >
    <nav
      class="navbar "
    >
      <div
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <!---->
           
          <a
            aria-label="Menu"
            class="navbar-burger has-text-white"
            href="#"
          >
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
          </a>
        </div>
         
        <div
          class="navbar-menu is-relative"
        >
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="control has-icons-right"
              >
                <label
                  class="is-sr-only"
                  for="search-form-input"
                >
                  Search
                </label>
                 
                <input
                  autocomplete="off"
                  class="input is-small search"
                  id="search-form-input"
                  placeholder="Search..."
                  type="search"
                />
                 
                <span
                  class="icon is-small is-right"
                >
                  <svg
                    class="feather feather-search"
                    fill="none"
                    height="24px"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </span>
              </div>
               
              <!---->
            </div>
             
            <a
              class="navbar-item has-text-white-ter"
              href="/rss"
            >
              <span
                class="icon"
              >
                <svg
                  class="feather feather-rss"
                  fill="none"
                  height="24px"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 11a9 9 0 0 1 9 9"
                  />
                  <path
                    d="M4 4a16 16 0 0 1 16 16"
                  />
                  <circle
                    cx="5"
                    cy="19"
                    r="1"
                  />
                </svg>
              </span>
               
              <span
                class="is-sr-only"
              >
                RSS
              </span>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
   
  <div
    class="hero-body fix-z-index"
  >
    <div
      class="container has-text-centered"
    >
      <h1
        class="title is-1 has-text-white"
      >
        
          title
        
      </h1>
       
      <h2
        class="subtitle has-text-white-bis"
      >
        
          description
        
      </h2>
       
    </div>
  </div>
   
  <!---->
</header>
`;

exports[`AppHeader override cover 1`] = `
<header
  class="hero is-medium fix-background"
  style="background-image: url(passCoverViaProps.png);"
>
  <div
    class="hero-head"
  >
    <nav
      class="navbar "
    >
      <div
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <!---->
           
          <a
            aria-label="Menu"
            class="navbar-burger has-text-white"
            href="#"
          >
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
          </a>
        </div>
         
        <div
          class="navbar-menu is-relative"
        >
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="control has-icons-right"
              >
                <label
                  class="is-sr-only"
                  for="search-form-input"
                >
                  Search
                </label>
                 
                <input
                  autocomplete="off"
                  class="input is-small search"
                  id="search-form-input"
                  placeholder="Search..."
                  type="search"
                />
                 
                <span
                  class="icon is-small is-right"
                >
                  <svg
                    class="feather feather-search"
                    fill="none"
                    height="24px"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </span>
              </div>
               
              <!---->
            </div>
             
            <a
              class="navbar-item has-text-white-ter"
              href="/rss"
            >
              <span
                class="icon"
              >
                <svg
                  class="feather feather-rss"
                  fill="none"
                  height="24px"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 11a9 9 0 0 1 9 9"
                  />
                  <path
                    d="M4 4a16 16 0 0 1 16 16"
                  />
                  <circle
                    cx="5"
                    cy="19"
                    r="1"
                  />
                </svg>
              </span>
               
              <span
                class="is-sr-only"
              >
                RSS
              </span>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
   
  <div
    class="hero-body fix-z-index"
  >
    <div
      class="container has-text-centered"
    >
      <h1
        class="title is-1 has-text-white"
      >
        
          
        
      </h1>
       
      <!---->
       
    </div>
  </div>
   
  <!---->
</header>
`;

exports[`AppHeader override default slot 1`] = `
<header
  class="hero is-medium fix-background"
  style="background-image: url(cover.png);"
>
  <div
    class="hero-head"
  >
    <nav
      class="navbar "
    >
      <div
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <!---->
           
          <a
            aria-label="Menu"
            class="navbar-burger has-text-white"
            href="#"
          >
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
          </a>
        </div>
         
        <div
          class="navbar-menu is-relative"
        >
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="control has-icons-right"
              >
                <label
                  class="is-sr-only"
                  for="search-form-input"
                >
                  Search
                </label>
                 
                <input
                  autocomplete="off"
                  class="input is-small search"
                  id="search-form-input"
                  placeholder="Search..."
                  type="search"
                />
                 
                <span
                  class="icon is-small is-right"
                >
                  <svg
                    class="feather feather-search"
                    fill="none"
                    height="24px"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </span>
              </div>
               
              <!---->
            </div>
             
            <a
              class="navbar-item has-text-white-ter"
              href="/rss"
            >
              <span
                class="icon"
              >
                <svg
                  class="feather feather-rss"
                  fill="none"
                  height="24px"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 11a9 9 0 0 1 9 9"
                  />
                  <path
                    d="M4 4a16 16 0 0 1 16 16"
                  />
                  <circle
                    cx="5"
                    cy="19"
                    r="1"
                  />
                </svg>
              </span>
               
              <span
                class="is-sr-only"
              >
                RSS
              </span>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
   
  <div
    class="hero-body fix-z-index"
  >
    default slot
  </div>
   
  <!---->
</header>
`;

exports[`AppHeader put footer slot 1`] = `
<header
  class="hero is-medium fix-background"
  style="background-image: url(cover.png);"
>
  <div
    class="hero-head"
  >
    <nav
      class="navbar "
    >
      <div
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <!---->
           
          <a
            aria-label="Menu"
            class="navbar-burger has-text-white"
            href="#"
          >
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
          </a>
        </div>
         
        <div
          class="navbar-menu is-relative"
        >
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="control has-icons-right"
              >
                <label
                  class="is-sr-only"
                  for="search-form-input"
                >
                  Search
                </label>
                 
                <input
                  autocomplete="off"
                  class="input is-small search"
                  id="search-form-input"
                  placeholder="Search..."
                  type="search"
                />
                 
                <span
                  class="icon is-small is-right"
                >
                  <svg
                    class="feather feather-search"
                    fill="none"
                    height="24px"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </span>
              </div>
               
              <!---->
            </div>
             
            <a
              class="navbar-item has-text-white-ter"
              href="/rss"
            >
              <span
                class="icon"
              >
                <svg
                  class="feather feather-rss"
                  fill="none"
                  height="24px"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 11a9 9 0 0 1 9 9"
                  />
                  <path
                    d="M4 4a16 16 0 0 1 16 16"
                  />
                  <circle
                    cx="5"
                    cy="19"
                    r="1"
                  />
                </svg>
              </span>
               
              <span
                class="is-sr-only"
              >
                RSS
              </span>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
   
  <div
    class="hero-body fix-z-index"
  >
    <div
      class="container has-text-centered"
    >
      <h1
        class="title is-1 has-text-white"
      >
        
          
        
      </h1>
       
      <!---->
       
    </div>
  </div>
   
  <div
    class="hero-foot fix-z-index"
  >
    footer
  </div>
</header>
`;

exports[`AppHeader with home button 1`] = `
<header
  class="hero is-medium fix-background"
  style="background-image: url(cover.png);"
>
  <div
    class="hero-head"
  >
    <nav
      class="navbar "
    >
      <div
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <div
            class="navbar-brand"
          >
            <a
              class="navbar-item has-text-white-bis"
            >
              
          title
        
            </a>
          </div>
           
          <a
            aria-label="Menu"
            class="navbar-burger has-text-white"
            href="#"
          >
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
             
            <span
              aria-hidden="true"
            />
          </a>
        </div>
         
        <div
          class="navbar-menu is-relative"
        >
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="control has-icons-right"
              >
                <label
                  class="is-sr-only"
                  for="search-form-input"
                >
                  Search
                </label>
                 
                <input
                  autocomplete="off"
                  class="input is-small search"
                  id="search-form-input"
                  placeholder="Search..."
                  type="search"
                />
                 
                <span
                  class="icon is-small is-right"
                >
                  <svg
                    class="feather feather-search"
                    fill="none"
                    height="24px"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="11"
                      cy="11"
                      r="8"
                    />
                    <line
                      x1="21"
                      x2="16.65"
                      y1="21"
                      y2="16.65"
                    />
                  </svg>
                </span>
              </div>
               
              <!---->
            </div>
             
            <a
              class="navbar-item has-text-white-ter"
              href="/rss"
            >
              <span
                class="icon"
              >
                <svg
                  class="feather feather-rss"
                  fill="none"
                  height="24px"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 11a9 9 0 0 1 9 9"
                  />
                  <path
                    d="M4 4a16 16 0 0 1 16 16"
                  />
                  <circle
                    cx="5"
                    cy="19"
                    r="1"
                  />
                </svg>
              </span>
               
              <span
                class="is-sr-only"
              >
                RSS
              </span>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
   
  <div
    class="hero-body fix-z-index"
  >
    <div
      class="container has-text-centered"
    >
      <h1
        class="title is-1 has-text-white"
      >
        
          
        
      </h1>
       
      <!---->
       
    </div>
  </div>
   
  <!---->
</header>
`;
